source/percentile.js

```diff
@@ -0,0 +1,56 @@
+import { curry, sort, length, filter} from "ramda"
+
+import isNumeric from "./internal/isNumeric"
+
+/**
+ * Percentile of a value in a series of series using midpoint interpolation. see https://numpy.org/doc/stable/reference/generated/numpy.percentile.html
+ *
+ * @func
+ * @memberOf Z
+ * @category Analysis
+ * @param {Number} percentile to calculate
+ * @param {Array} array of values to calculate percentile from 
+ * @return {Number}
+ * @example
+ *
+ * const series = [7, 2, 30, 56, 75, 150]
+ * Z.percentile(95, series)
+ * // 112.5
+ */
+const percentile = curry((in_p, arr ) => {
+
+    const p = confirmPercentile(in_p)
+    const sortedArr = sort(diff, filter(isNumeric, arr))
+
+    // if requesting 0 or 1 do the easy thing. 
+    if (in_p == 0) {
+        return sortedArr[0];
+    }
+    else if (in_p == 1) {
+        return sortedArr[-1];
+    }
+
+    const i = p * length(sortedArr)
+    var low_index = sortedArr[Math.floor(i - 1)]
+    var high_index = sortedArr[Math.floor(i)]
+    return (low_index + high_index)/2
+
+
+    
+})
+
+const diff = function(a, b) { return a - b; };
+
+function confirmPercentile(num) {
+    if (num <1 & num >= 0) {
+        return num
+    }
+    else if (num >=1 & num <= 100) {
```

Comment:
Thanks for this! Definitely an important addition. 

I think it's worth deciding whether to require percentiles of the form 0<=p<=100 or 0<=p<=1.

The pandas quantile function does the latter, whereas the np percentiles function takes the former.

I don't have a view which one to choose, but handling both seems like it could cause confusion. Eg here, it's not obvious to the user whether p=1 is requesting the 1st %ile  or the 100th %ile. (In fact, if I'm reading this correctly, `p` becomes 0.01, but then `sortedArr[-1]` gets returned.)